src/app/guards/no-auth.guard.spec.ts

```diff
@@ -0,0 +1,40 @@
+import { TestBed, inject } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { NoAuthGuard } from './no-auth.guard';
+import { AuthService } from '../services/auth.service';
+
+class MockRouter {
+	navigate(path) {}
+}
+
+describe('NoAuthGuard', () => {
+	let authGuard: NoAuthGuard;
+	let MockAuthService;
+	let router;
+	beforeEach(() => {
+		TestBed.configureTestingModule({
+			imports: [ RouterTestingModule ],
+			providers: [ NoAuthGuard, { provide: { AuthService, useValue: MockAuthService } } ]
+		});
+		router = new MockRouter();
+		MockAuthService = jasmine.createSpyObj([ 'isLoggedIn' ]);
+		authGuard = new NoAuthGuard(MockAuthService, router);
+	});
+
+	it(
+		'should ...',
```

Comment:
Hey @CryceTruly, I am not sure what this is dong here.
can you shine some more light to it